common/mcp_tool_call_conn.py

```
@@ -76,7 +76,12 @@ async def _mcp_server_loop(self) -> None:
         for h, v in raw_headers.items():
             nh = Template(h).safe_substitute(self._server_variables)
             nv = Template(v).safe_substitute(self._server_variables)
-            if nh.strip() and nv.strip().strip("Bearer"):
+            # `.strip("Bearer")` strips the character *set* {B,e,a,r}, so a
+            # header whose value is made only of those letters -- "rare", "rae" --
+            # collapsed to "" and the header was silently dropped. The intent is
+            # to skip an Authorization header carrying no token after "Bearer".
+            token = nv.strip()
+            if nh.strip() and token and token != "Bearer":
```

Ruff (0.16.2)

[error] Possible hardcoded password assigned to: "token"

(S105)